Create compatible-apiModule.apiAction-combinations

gillnd opened the pull request:
Added a new page for compatible apiModule and apiAction combinations for API.getMetadata, API.getProcessedReport, API.getRowEvolution, API.ImageGraph.get

### Description:

Please include a description of this change and which issue it fixes. If no issue exists yet please include context and what problem it solves.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

gillnd committed:
Create compatible-apiModule.apiAction-combinations

Added a new page for compatible apiModule and apiAction combinations for API.getMetadata, API.getProcessedReport, API.getRowEvolution, API.ImageGraph.get

michalkleiner commented on docs/5.x/compatible-apiModule.apiAction-combinations:
```diff
@@ -0,0 +1,802 @@
+- `Contents.getContentNames`: Returns the names of tracked content blocks (e.g. banners), including impressions, interactions, and interaction rate.
+- `Contents.getContentPieces`: Returns data on specific pieces of content (e.g. images or text), including how often each piece was viewed or clicked.
+
+## Crash Analytics
```
@gillnd should the headings here and further down until (and including) `## Resolution` be `###` level?

gillnd replied:
@michalkleiner Corrected the heading levels.

michalkleiner commented on docs/5.x/compatible-apiModule.apiAction-combinations:
```diff
@@ -0,0 +1,802 @@
+
+- `VisitsSummary.get`: Displays a graph about the report showing general visit metrics including visits, unique visitors, and bounce rate.
+
+
```
Can probably remove one or both trailing lines.

michalkleiner commented on docs/5.x/compatible-apiModule.apiAction-combinations:
```diff
@@ -0,0 +1,802 @@
+With the [Matomo Reporting API](https://developer.matomo.org/api-reference/reporting-api), you can query specific reports using a valid combination of an `apiModule` and `apiAction` parameter to identify the exact dataset to retrieve. The `apiModule` specifies the plugin or area of functionality (such as Actions, Goals, 
+or Referrers), while apiAction refers to the specific method exposed by that module (such as `getPageUrls`, `getConversions`, or `getCountry`). Together, these define the API method you want to call. 
+
+Four Matomo API methods support the combined used of `apiModule` and `apiAction` parameters:
```
```suggestion
Four Matomo API methods support the combined use of `apiModule` and `apiAction` parameters:
```

michalkleiner commented on docs/5.x/compatible-apiModule.apiAction-combinations:
```diff
@@ -0,0 +1,802 @@
+---
+category: API Reference
+title: Compatible apiModule and apiAction combinations
+---
```
Looking at how other files are structured, this might be a better way for consitency.

```suggestion
---
category: API Reference
---
# Compatible apiModule and apiAction combinations
```

It would also mean we need to move all subheadings by a level.

michalkleiner replied:
@gillnd we need to move all the other headings by a level, we should only have one `#` heading level within the document, the rest should be `##`, `###` and `####` respectively.

gillnd replied:
@michalkleiner Okay ok got. Fixed the headings

michalkleiner reviewed (changes requested):
Nice job @gillnd, left a few minor comments.

gillnd committed:
Update docs/5.x/compatible-apiModule.apiAction-combinations

Co-authored-by: Michal Kleiner <mk@011.nz>

gillnd committed:
Update docs/5.x/compatible-apiModule.apiAction-combinations

Co-authored-by: Michal Kleiner <mk@011.nz>

gillnd committed:
Update compatible-apiModule.apiAction-combinations

Resolved Michal's comments on the heading levels and trailing lines at the end.

gillnd committed:
Update compatible-apiModule.apiAction-combinations
